class-tgm-plugin-activation.php

```diff
@@ -2023,12 +2064,31 @@ public function force_activation() {
 					} elseif ( $this->can_plugin_activate( $slug ) ) {
 						// There we go, activate the plugin.
 						activate_plugin( $plugin['file_path'] );
+						$this->append_transient( 'tgmpa_force_activated_plugins', $slug );
 					}
 				}
 			}
 		}
 
 		/**
+		 * Set/update a transient, appending the new value if the transient already exists.
+		 *
+		 * @since 2.6.0
+		 *
+		 * @param string       $transient_key Name of transient.
+		 * @param string|array $data To set or add to transient.
+		 */
+		protected function append_transient( $transient_key, $data ) {
+			if ( $transient_data = get_transient( $transient_key ) ) {
+				(array) $transient_data[] = $data;
+			} else {
+				$transient_data = (array) $data;
+			}
+
+			set_transient( $transient_key, $transient_data );
+		}
```

Comment:
Just wondering if we can make this slightly more efficient - if there are five plugins being force activated, the transient (and the associated cache) would be (re)set five times or am I missing something ? Could we collect the values and set the transient in one go ? or would that become very complicated ?